modules/cnf-performing-platform-verification-latency-tests.adoc

```diff
@@ -0,0 +1,756 @@
+
+. Optional: To run the latency tests against a specific performance profile, run the following command, substituting appropriate values:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e LATENCY_TEST_RUNTIME=600 -e OSLAT_MAXIMUM_LATENCY=20 \
+-e PERF_TEST_PROFILE=<performance_profile> registry.redhat.io/openshift4/cnf-tests-rhel8:v{product-version} \
+/usr/bin/test-run.sh -ginkgo.focus="\[performance\]\[config\]|\[performance\]\ Latency\ Test"
+----
++
+where:
++
+--
+<performance_profile> :: Is the name of the performance profile you want to run the latency tests against.
+--
+
+[id="cnf-performing-end-to-end-tests-running-hwlatdetect"]
+=== Running hwlatdetect
+
+The `hwlatdetect` tool is available in the `rt-kernel` package with a regular subscription of Red Hat Enterprise Linux 8.
+
+.Prerequisites:
+
+* You have installed the real-time kernel in the cluster
+
+* You have logged in to `registry.redhat.io` with your Customer Portal credentials
+
+.Procedure
+
+* To run the `hwlatdetect` tests, run the following command, substituting variable values as appropriate:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e DISCOVERY_MODE=true -e ROLE_WORKER_CNF=worker-cnf \
+-e LATENCY_TEST_CPUS=600 -e LATENCY_TEST_RUNTIME=600 \
+-e MAXIMUM_LATENCY=20 registry.redhat.io/openshift4/cnf-tests-rhel8:v{product-version} \
+/usr/bin/test-run.sh -ginkgo.v -ginkgo.focus="hwlatdetect"
+----
++
+`hwlatdetect` runs for 10 minutes (600 seconds). The test runs successfully when the maximum observed latency is lower than `MAXIMUM_LATENCY` (20 μs).
++
+[NOTE]
+====
+For valid results, the test should run for at least 12 hours.
+====
++
+If the results exceed the latency threshold, the test fails.
++
+.Example failure output
+[source,terminal]
+----
+running /usr/bin/cnftests -ginkgo.v -ginkgo.focus=hwlatdetect
+I0908 15:25:20.023712      27 request.go:601] Waited for 1.046586367s due to client-side throttling, not priority and fairness, request: GET:https://api.hlxcl6.lab.eng.tlv2.redhat.com:6443/apis/imageregistry.operator.openshift.io/v1?timeout=32s
+Running Suite: CNF Features e2e integration tests
+=================================================
+Random Seed: 1662650718
+Will run 1 of 194 specs
+
+[...]
+
+• Failure [283.574 seconds]
+[performance] Latency Test
+/remote-source/app/vendor/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency/latency.go:62
+  with the hwlatdetect image
+  /remote-source/app/vendor/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency/latency.go:228
+    should succeed [It]
+    /remote-source/app/vendor/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency/latency.go:236
```

Comment:
Note that this is relevant for 4.11/4.12 releases, where the tests exist in NTO. for other versions please consider using the old command outputs as per each version documentation considering the availability of the tools in each version (hwlat and cyclictest are available since 4.9)

Reply:
Cool. Let's get this merged to main/enterprise-4.11+, and then I will manually CP and adjust for previous versions.